### PR TITLE
[Testcase Refactoring] Decouple test_fsdp_checkpoint from hard-coded devices and label ACCELERATOR classes

### PR DESCRIPTION
**Pull request by twthighd**
## Summary

This PR labels `TestFSDPCheckpoint` and `TestFSDPCheckpointSubmodule` as `HardwareClassification.ACCELERATOR`, switches both to `instantiate_device_type_tests(except_for=("cpu",))`, and completes the device-parameter migration in the test bodies — including using `torch.device(device).type` (type-only) for FSDP `device_id` and all model/tensor creation so each rank resolves to its own device instead of degenerating to device 0.

## Changes

- **`test/distributed/fsdp/test_fsdp_checkpoint.py`**:
  - Added `hw_classification = HardwareClassification.ACCELERATOR` to both `TestFSDPCheckpoint` and `TestFSDPCheckpointSubmodule`, plus the `HardwareClassification` import. Both classes' `test_*` methods require an accelerator (`@skip_if_lt_x_gpu(2)`, `device` param) and exercise FSDP checkpoint behavior on the device-tensor path.
  - Switched `TestFSDPCheckpoint` from `@instantiate_parametrized_tests` to `instantiate_device_type_tests(except_for=("cpu",), allow_xpu=True)`, and changed `TestFSDPCheckpointSubmodule`'s `only_for=("cuda","hpu","xpu")` to `except_for=("cpu",)` (dropping the hard-coded `devices` variable). This aligns with the TEST_LINTER direction (PR #190173): ACCELERATOR classes must NOT use `only_for`.
  - Migrated every test method to the injected `device` parameter: added `device` to both class-1 methods and to `SequentialModule.__init__`, deleted the module-level `device_type = torch.device(get_devtype())` and the `get_devtype` import, and replaced all `device_type.type` / `self.device_type` references in the bodies with `device`.
  - Replaced 4 `TestFSDPCheckpoint.SequentialModule(...)` references with `self.SequentialModule(...)`. `instantiate_device_type_tests` removes the original class name from scope (replaces it with per-device variants), so a test body referencing the original class name raises `NameError` at runtime; the variants inherit from the original class, so `self.SequentialModule` resolves correctly. This was latent until the c10d backend-type lookup for PrivateUse1 was fixed (#187650 / Ascend/pytorch#3947), because workers previously exited at `dist.barrier(device_ids=)` before reaching this code.
  - Used `torch.device(device).type` (type-only) for FSDP `device_id`, `nn.Linear().to(...)`, `torch.randn(device=...)`, and `SequentialModule(device=...)`. The injected `device` parameter is indexed (e.g. `"npu:0"`); passing it bare to FSDP `device_id` forces all ranks to device 0 (single-card degeneration — no error but loses multi-card verification). `torch.device(device).type` strips the index, so FSDP's `_get_device_from_device_id` resolves each rank to its own current device (true multi-card distribution).

## Motivation

`TestFSDPCheckpoint` / `TestFSDPCheckpointSubmodule` had no `hw_classification` (silently dropped under `--hw-classification ACCELERATOR` filtering) and used `only_for=("cuda","hpu","xpu")`, excluding PrivateUse1-based out-of-tree accelerators. The test bodies read a module-level `device_type` constant — a single value that does not change across per-device variants — which under `instantiate_device_type_tests` would use the wrong device on non-default variants. Additionally, passing the indexed `device` directly to FSDP `device_id` silently degenerated multi-rank tests to single-card, masking real multi-card behavior.

## Test Plan

- Verified the FSDP test infra routes to the accelerator backend (not a gloo/CPU false pass):

```
python -c "from torch.testing._internal.common_fsdp import DEVICE_TYPE, DISTRIBUTED_BACKEND, DEVICE_COUNT; print(DEVICE_TYPE, DISTRIBUTED_BACKEND, DEVICE_COUNT)"
# <device> <backend> <count>  (e.g. an accelerator backend, not cpu gloo 1)
```

- End-to-end on an 8-card accelerator backend (with the `common_fsdp.py` accelerator branch from #187650 applied):

```
python test/distributed/fsdp/test_fsdp_checkpoint.py --hw-classification ACCELERATOR
# 17/17 pass; FSDP warnings confirm per-rank device resolution
# (rank 0 -> device 0, rank 3 -> device 3, ...) — true multi-card, not single-card degeneration
```

- `lintrunner` clean; `review-test-refactoring` skill review: no findings.

## Notes

- Both classes retain `@skip_if_lt_x_gpu(2)` unchanged. Making `skip_if_lt_x_gpu` recognize out-of-tree accelerators is a separate framework-level change to `common_distributed.py`, tracked in pytorch/pytorch#187650; this PR does not modify that helper.
- This is part of the test case refactoring initiative tracked in #185590.

### CODE REVIEW

**Comment by twthighd** [comment]
@wjlFlyer @pjyFight @FuDdd @fffrog @lyriexs666 @jishuangfeng @JamesD18 @JiasenTian

**Review by wjlFlyer — APPROVED** [review]
